packages/core/src/controllers/SwapController.ts

```diff
@@ -464,14 +466,30 @@ export const SwapController = {
     }
 
     state.loading = true
-    state.toTokenAmount = SwapCalculationUtil.getToTokenAmount({
-      sourceToken: state.sourceToken,
-      toToken: state.toToken,
-      sourceTokenPrice: state.sourceTokenPriceInUSD,
-      toTokenPrice: state.toTokenPriceInUSD,
-      sourceTokenAmount: state.sourceTokenAmount
+
+    const amountDecimal = NumberUtil.bigNumber(state.sourceTokenAmount).multipliedBy(
+      10 ** sourceToken.decimals
+    )
+
+    const quoteResponse = await BlockchainApiController.fetchSwapQuote({
+      userAddress: address,
+      projectId: OptionsController.state.projectId,
+      from: sourceToken.address,
+      to: toToken.address,
+      gasPrice: state.gasFee,
+      amount: amountDecimal.toString()
     })
 
+    if (!quoteResponse?.quotes?.[0]) {
+      return
+    }
+
+    const toTokenAmount = NumberUtil.bigNumber(quoteResponse?.quotes?.[0].toAmount)
+      .dividedBy(10 ** toToken.decimals)
```

Comment:
is there an issue with dividing and getting a decimal here ? (since we are dealing with bignumbers)

Reply:
Nope - this is an expected calculation that we are multiplying or dividing the amount values with the decimal value of the related token, we need to do this to make BigInt to Float